packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,306 @@
+  removeSuccessMessage,
+  successMessage,
+  disabled,
+  dragAndDrop,
+  className = '',
+  errorText,
+  helperText,
+  onChange,
+  required,
+  style,
+  accept,
+  multiple,
+}: FileInputProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [selectedFiles, setSelectedFiles] = useState<File[]>([]);
+  const [successText, setSuccessText] = useState<string | undefined>();
+  const hasFilesSelected = selectedFiles && selectedFiles.length > 0;
+  const buttonId = `${id}-button`;
+  const fileListId = `${id}-list`;
+  const hasDragAndDrop = !!dragAndDrop;
+  const dropAreaRef = useRef<HTMLDivElement>(null);
+  const [isDragOverDrop, setIsDragOverDrop] = useState<boolean>(false);
+
+  const wrapperProps = {
+    className,
+    helperText,
+    successText,
+    errorText,
+    id,
+    label,
+    required,
+    style,
+  };
+
+  const passButtonClickToInput = () => {
+    if (inputRef.current) {
+      inputRef.current.click();
+    }
+  };
+
+  const resetFileInputValue = () => {
+    if (inputRef.current) {
+      inputRef.current.value = '';
+    }
+  };
+
+  const handleSingleFileChange = (files: File[]) => {
+    if (files.length > 0) {
+      setSelectedFiles(files);
+      setSuccessText(successMessage);
+    } else {
+      setSuccessText(undefined);
+    }
+  };
+
+  const handleMultipleChange = (files: File[]) => {
+    if (files.length > 0) {
+      const [replacedFiles, newFiles] = files.reduce(
+        (acc: [File[], File[]], file: File) => {
+          if (findDuplicateByNameAndType(selectedFiles, file)) {
+            return [[...acc[0], file], acc[1]];
+          }
+          return [acc[0], [...acc[1], file]];
+        },
+        [[], []],
+      );
+
+      const selectedWithoutReplacedFiles = selectedFiles.filter(
+        (selectedFile: File) => !findDuplicateByNameAndType(replacedFiles, selectedFile),
+      );
+      setSelectedFiles([...selectedWithoutReplacedFiles, ...replacedFiles, ...newFiles]);
+      setSuccessText(successMessage);
+    } else {
+      setSuccessText(undefined);
+    }
+  };
+
+  const onFilesChange = (files: File[]) => {
+    if (multiple) {
+      handleMultipleChange(files);
+    } else {
+      handleSingleFileChange(files);
+    }
+  };
+
+  const removeFileFromList = (fileToRemove: File) => {
+    const selectedFilesWithoutRemoved = selectedFiles.filter(
+      (file: File) => !isEqualFileBy(['name', 'type', 'size', 'lastModified'], file, fileToRemove),
+    );
+    setSelectedFiles(selectedFilesWithoutRemoved);
+    setSuccessText(removeSuccessMessage);
+  };
+
+  const onDragEnter = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(true);
+  };
+
+  const onDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(false);
+  };
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedFiles);
```

Comment:
I guess this means that the onChange is also run after the component is mounted, before the user has selected any files? Just wondering if this behaviour is expected. What if user only expects the onChange to be run when there's actual change in the input?

Reply:
The onChange is also called when a file is removed from the list or dragged into the drag-area. 
The selectedFiles value might differ from the value of the actual input.
But it might not be necessary to run this effect when the component has mounted.

Reply:
Good point! And the users of the onChange method should always check if there are any selected files. Though it might be little unexpected that the onChange will be called on first render even there's no change.

Reply:
- [ ] prevent initial onChange